tests/core/test_filter_pred.py

```diff
@@ -0,0 +1,88 @@
+import asyncio
+
+import pytest
+
+from conftest import (
+    generate_unique_name,
+)
+from lahja import (
+    BaseEvent,
+    ConnectionConfig,
+    Endpoint,
+)
+
+
+@pytest.mark.asyncio
+async def test_subscribed_events(event_loop: asyncio.AbstractEventLoop) -> None:
+    class TestSubscriptionEvent(BaseEvent):
+        pass
+
+    endpoint = Endpoint()
+    endpoint.subscribe(TestSubscriptionEvent, lambda event: None)
+
+    assert endpoint.subscribed_events == {TestSubscriptionEvent}
+
+    class TestStreamEvent(BaseEvent):
+        pass
+
+    async def stream() -> None:
+        async for _ in endpoint.stream(TestStreamEvent):  # noqa: F841
+            pass
+
+    future = asyncio.ensure_future(stream())
+    await asyncio.sleep(0)  # give stream a chance to be scheduled
+
+    assert endpoint.subscribed_events == {TestSubscriptionEvent, TestStreamEvent}
+
+    future.cancel()
+    endpoint.stop()
+
+
+class TestEvent(BaseEvent):
+    # This must be importable in order for it to be pickleable
+    pass
+
+
+@pytest.mark.asyncio
+async def test_remote_subscribed_events(event_loop: asyncio.AbstractEventLoop) -> None:
+    local = Endpoint()
+    local.subscribe(TestEvent, lambda event: None)
+
+    assert local.subscribed_events == {TestEvent}
+
+    local_config = ConnectionConfig.from_name(generate_unique_name())
+    await local.start_serving(local_config, event_loop)
+
+    remote = Endpoint()
+    remote._loop = event_loop  # Endpoint assumes you'll serve before connecting
+    await remote.connect_to_endpoints(local_config)
+
+    proxy = remote._connected_endpoints[local_config.name]
+    assert proxy.proxy.subscribed_events() == {TestEvent}
+
+    remote.stop()
```

Comment:
This looks like something that should be doable via a context manager or in a `try/finally`.  What happens if we forget to explicitly stop these or the assertions fail and these statements don't get hit?

Reply:
Just wrapped it into a context manager.